internal/mode/static/state/graph/gatewayclass.go

```diff
@@ -76,7 +75,7 @@ func buildGatewayClass(gc *v1beta1.GatewayClass, npCfg *ngfAPI.NginxProxy) *Gate
 	}
 }
 
-func validateGatewayClass(gc *v1beta1.GatewayClass, npCfg *ngfAPI.NginxProxy) error {
+func validateGatewayClass(gc *v1beta1.GatewayClass, npCfg *NginxProxy) error {
 	if gc.Spec.ParametersRef != nil {
```

Comment:
Can we add validation for the new fields of NginxProxy?

While those validated by CRD validation, that validation is not guaranteed to be running (could be changed or have a bug, for example)

To prevent malicious value in NGINX config, for other resources, we run additional validation, as documented here https://github.com/nginxinc/nginx-gateway-fabric/blob/main/design/resource-validation.md#ngf-specific-validation 

similar validation is done here https://github.com/nginxinc/nginx-gateway-fabric/tree/9c11e1e3fed84a9cfa71f3e4c61864e2e889c8ef/internal/mode/static/nginx/config/validation

Reply:
A related question about validation: If tracing is disabled, but a tracing endpoint is provided, what should we do?

Reply:
> A related question about validation: If tracing is disabled, but a tracing endpoint is provided, what should we do?

That's an accepted configuration, as it's supported by NGINX. The generated config looks like:

```
otel_exporter {
    endpoint simplest-collector.default.svc.cluster.local:4317;
    interval 5s;
    batch_size  512;
    batch_count 4;
}
otel_trace off;
otel_service_name default/gateway:ngf;
```